Description: X can abort if multiple video devices are present, and none are marked as primary.  This makes X just pick the first one it sees and carry on.
Author: Bryce Harrington <bryce@ubuntu.com>
Bug-Ubuntu: https://launchpad.net/bugs/459512

diff --git a/hw/xfree86/common/xf86pciBus.c b/hw/xfree86/common/xf86pciBus.c
index e52f1da..462faa9 100644
--- a/hw/xfree86/common/xf86pciBus.c
+++ b/hw/xfree86/common/xf86pciBus.c
@@ -135,8 +135,8 @@ xf86PciProbe(void)
                 }
                 else {
                     xf86Msg(X_NOTICE,
-                            "More than one possible primary device found\n");
-                    primaryBus.type ^= (BusType) (-1);
+                            "More than one possible primary device found.  Using first one seen.\n");
+                    break;
                 }
             }
         }
